.github/workflows/format.yml

```diff
@@ -0,0 +1,29 @@
+on:
+  push:
+  pull_request:
+
+name: "Format checks"
+permissions: {}
+
+jobs:
+  shellcheck:
+    name: Shellcheck
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run ShellCheck
+        uses: ludeeus/action-shellcheck@master
```

Comment:
I'm not super happy with running an arbitrarily changing action with an arbitrarily changing version of shellcheck.
Using the shellcheck executable that's available out of the box in GHA or Ubuntu would be slightly better. Or ideally this would be a precise version of a GitHub release of shellcheck, with its version managed by https://docs.renovatebot.com/modules/datasource/github-releases/

Reply:
Could be done in a follow-up, I suppose  🤔

Reply:
That's a good point. I just picked this up as it's an easy solution.
However, we can at least pin the version of the action.

I'm happy to use any other mechanism.

We could also simply use `grep -l -E '^#!(/usr/bin/env bash|/bin/sh|/bin/bash)' $(git ls-files)` to detect changed script files.

Reply:
Integration with pre-commit would also be nice 🤷